File: archive/o/octave/fibonacci.m
function fibonacci()
    usage = 'Usage: please input the count of fibonacci numbers to output';
    arg_list = argv();
    nargin = length(arg_list);
    if nargin == 0
        disp(usage);
        return;
    end

    n = str2num(arg_list{1});
    if length(n) ~= 1 || mod(n, 1) ~= 0
        disp(usage);
        return;
    end

    a(1) = 1;
    a(2) = 1;

    for i=1:n
        a(i+2) = a(i+1) + a(i);
        printf('%d: %d\n', i, a(i));
    end
end

File: archive/e/elixir/fibonacci.ex
defmodule Fibonacci do
  @doc """
  Returns the first `number` elements of the Fibonacci sequence as a list.
  """  
  @spec fibonacci(number :: integer()) :: list(integer())
  def fibonacci(number) do
    Stream.unfold({1, 1}, fn {m, n} -> {m, {n, m+n}} end)
    |> Enum.take(number)
  end
end

with [number] <- System.argv(),
     {number, ""} <- Integer.parse(number) do
  Fibonacci.fibonacci(number)
  |> Enum.with_index(1)
  |> Enum.each(fn {n, i} -> IO.puts("#{i}: #{n}") end)
else
  _ -> IO.puts("Usage: please input the count of fibonacci numbers to output")
end
